Adding comment for the function runFromCommandLine

abhishekgite446 opened the pull request:
Addressing [Issue 76](https://github.com/electron/fiddle-core/issues/76)

Adding comment using JSDoc for the function runFromCommandLine and replacing undocumented comments in etc/fiddle-core.api.md using `yarn docs`

abhishekgite446 commented:
Hi @dsanders11, apologies that I had to close my previous [commit](https://github.com/electron/fiddle-core/pull/82). Opening this new PR to address [Issue 76](https://github.com/electron/fiddle-core/issues/76). I have only update the description for the function `runFromCommandLine` for now, can you please have a look ?

dsanders11 commented:
It seems unnecessary to bump the version of `@microsoft/api-extractor`? The changes to `etc/fiddle-core.api.md` are also not included in these changes (which causes the workflow failure). Also, the comment shouldn't mention `runFromCommandLine`, and doesn't currently pass lint.

abhishekgite446 commented:
Thanks for having a look @dsanders11. I guess pushed the changes in a rush. Thanks a lot for spotting everything, I have addressed all the issues you have raised previously. Can you please have a look when you get time :)

abhishekgite446 commented:
Just so that you are aware, the only reason I bumped the version for `@microsoft/api-extractor` because I saw the following message when I executed `yarn docs`. Nevertheless the version bump was not required in the first place.

abhishekgite446 commented:
@dsanders11 when you get time can you please have a look :)

dsanders11 commented on src/installer.ts:
```diff
@@ -71,6 +91,7 @@ export class Installer extends EventEmitter {
     this.rebuildStates();
   }
 
+  /** Executes the subpath based on the platform */
```
```suggestion
  /** Returns the executable subpath based on the platform */
```

abhishekgite446 replied:
Done 👍

dsanders11 commented on src/versions.ts:
```diff
@@ -57,19 +59,22 @@ export interface Versions {
 
-  /** @returns true iff `version` is a release that this object knows about */
+  /** Returns true iff `version` is a release that this object knows about */
   isVersion(version: SemOrStr): boolean;
 
-  /** @returns all versions matching that major number. Sorted in branch order. */
+  /** Returns all versions matching that major number. Sorted in branch order. */
   inMajor(major: number): SemVer[];
 
-  /** @returns all versions in a range, inclusive. Sorted in branch order. */
+  /** Return all versions in a range, inclusive. Sorted in branch order. */
   inRange(a: SemOrStr, b: SemOrStr): SemVer[];
 
-  /** @returns {@link ReleaseInfo} iff `version` is a release that this object knows about */
+  /** Retrieves the release info */
   getReleaseInfo(version: SemOrStr): ReleaseInfo | undefined;
 }
```
It looks like these are marked as undocumented due to https://github.com/microsoft/rushstack/issues/3462 - I'd prefer to leave them as they were, the upstream bug will (hopefully) be fixed one day.

abhishekgite446 replied:
Fair enough. Done 👍

dsanders11 commented on src/command-line.ts:
```diff
@@ -5,6 +5,12 @@ import { ElectronVersions } from './versions';
+ * executes specific commands based on the arguments provided.
+ * It logs debug information and exits the process if invalid parameters are detected.
+ */
+
```
Remove the extra blank line here.

abhishekgite446 replied:
Done 👍

dsanders11 commented on src/fiddle.ts:
```diff
@@ -55,6 +65,12 @@ export class FiddleFactory {
+   * checking out a specified branch (default: 'master'),
+   * and setting the main file path based on the cloned files.
+   */
+
```
Remove the extra blank line here.

abhishekgite446 replied:
Done 👍

dsanders11 commented on src/fiddle.ts:
```diff
@@ -74,6 +90,11 @@ export class FiddleFactory {
+   * This method creates a Fiddle instance by saving a collection of filename-content pairs to a temporary directory
+   * and setting the main file path accordingly.
+   */
+
```
Remove the extra blank line here.

abhishekgite446 replied:
Done 👍

abhishekgite446 commented on src/paths.ts:
```diff
@@ -1,22 +1,26 @@
 import envPaths from 'env-paths';
 
+/**
+ * Represents a set of paths used by an application.
```
Done 👍

abhishekgite446 commented:
@dsanders11 as suggested made all changes. Please have a look at your convenience.

dsanders11 commented on src/fiddle.ts:
```diff
@@ -12,12 +12,16 @@ function hashString(str: string): string {
   constructor(
+    /** It serves as the entry point or the primary script file for the fiddle */
     public readonly mainPath: string, // /path/to/main.js
+    /** It represents the code */
```
`source` is the source of the fiddle (URL, file path, etc)

abhishekgite446 replied:
Done 👍

dsanders11 reviewed (commented):
Let's avoid starting the comments with wordings like "Represents", "Overrides", "This method", "This object" and "This function". I'd say avoid the usage of "represents" all together, similarly with "it" and "this" when possible.

For getter methods (like those in `src/versions.ts`) drop the starting "Gets", document it like it's a variable, rather than a method.

abhishekgite446 commented on src/fiddle.ts:
```diff
@@ -55,6 +65,11 @@ export class FiddleFactory {
 
+  /**
+   * This method creates a Fiddle instance by cloning a Git repository into a temporary directory,
+   * checking out a specified branch (default: 'master'),
```
Done 👍

abhishekgite446 commented on src/installer.ts:
```diff
@@ -22,30 +23,49 @@ export type ProgressObject = { percent: number };
 }
 
+/**
+ * Represents an event that indicates the installation state of something.
```
Done 👍

abhishekgite446 commented:
@dsanders11 as suggested I have amended the comments, please have a look at your convenience 👍
